server/src/core/agent/agent_service.py

```diff
@@ -64,13 +64,15 @@ def handle_user_input(self, req: UserInputDto) -> dict:
         self.history.add_message(add_message_dto)
         is_action = self.is_action(req)
 
-        if self.cache.exists(req.question):
-            message = self.cache.get(req.question)
+        # TODO: Eliminate `cache.exists` call:
```

Comment:
Remove cache exists? why?

Reply:
Because using one `self.cache.get(req.question)` with `!= None` is more efficient than using both `self.cache.exists` and `self.cache.get`.